sdk_extensions/logging_support/src/main/java/io/opentelemetry/logging/SizeOrLatencyBatchStrategy.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.logging;
+
+import io.opentelemetry.logging.api.LogRecord;
+import java.lang.ref.WeakReference;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nullable;
+
+/**
+ * Batching strategy that allows specifying both a maximum size and a maximum amount of time before
+ * sending a batch of records. Batch will be flushed whenever the list of records queued exceeds the
+ * maximum threshold (default 50), or a set timeout expires (default 5 seconds).
+ */
+public class SizeOrLatencyBatchStrategy implements LoggingBatchStrategy {
+  private static final int DEFAULT_BATCH_SIZE = 50;
+  private static final int DEFAULT_MAX_DELAY = 5;
+  private static final TimeUnit DEFAULT_MAX_DELAY_UNITS = TimeUnit.SECONDS;
+  private static final Timer timer = new Timer();
+
+  private final int maxBatch;
+  private final int maxDelay;
+  private final TimeUnit maxDelayUnits;
+
+  private WeakReference<LoggingBatchExporter> batchHandler;
+  private List<LogRecord> batch = new ArrayList<>();
```

Comment:
Can this be implemented with a Queue, rather than a bare List? The synchronization needed to manage this seems like a recipe for bad contention issues.